assets/tests/hooks/useNearestIntersection.test.ts

```diff
@@ -40,3 +47,81 @@ describe("useNearestIntersection", () => {
     expect(mockNearestIntersection).toHaveBeenCalledTimes(1)
   })
 })
+
+describe("useNearestIntersectionResult", () => {
+  const renderUseNearestIntersectionResult = (location: LatLngLiteral) =>
+    renderHook(({ lat, lng }) => useNearestIntersectionResult(lat, lng), {
+      initialProps: location,
+    })
+
+  test("when api call returns null, should return error", async () => {
+    ;(Api.fetchNearestIntersection as jest.Mock).mockReturnValueOnce(
+      Promise.resolve(null)
+    )
+    const location = localLatLngFactory.build()
+
+    const { result } = renderUseNearestIntersectionResult(location)
+
+    await waitFor(() => expect(result.current).toEqual(Err(expect.anything())))
+  })
+
+  test("when first rendered, should return loading state, then resolve to intersection", async () => {
+    const intersection = gridIntersectionFactory.build()
+    ;(Api.fetchNearestIntersection as jest.Mock).mockReturnValueOnce(
+      Promise.resolve(intersection)
+    )
+
+    const location = localLatLngFactory.build()
+    const { result, rerender } = renderUseNearestIntersectionResult(location)
+
+    expect(result.current).toEqual(Loading())
+
+    await act(async () => rerender(location))
+
+    expect(result.current).toEqual(Ok(intersection))
+  })
+
+  // Should it retry if the last result was an error?
```

Comment:
**todo:** if we do want it to behave this way, let's create a followup ticket and remove this comment, just because we'll probably lose track if the comment is the only thing documenting that we wanted to do that.

Reply:
I'm not sure we should, the old code doesn't, and I'm not quite sure what edge cases cause the geocoding api to return errors. 
Do you think we should be retrying on error?